x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -6,13 +6,12 @@
 
 import turf from 'turf';
 import turfBooleanContains from '@turf/boolean-contains';
-
-import { GIS_API_PATH } from '../../common/constants';
 import {
   getLayerList,
   getLayerListRaw,
   getDataFilters,
   getSelectedLayer,
+  getSelectedLayerId,
```

Comment:
There are several action creators that use `const selectedLayer = getSelectedLayer(getState())` just to later call `selectedLayer.getId()`. Would you mind replacing these instances with this new selector?

Reply:
:+1:

Reply:
In the cases covered by this PR, I'd say we don't need to pass through the selected layer id at all. We can get that in the store directly and save a round trip. Getting the selected layer is only really useful if we're doing something "side-effecty" like layer data sync.